### PR TITLE
Swift 3 Support

### PR DESCRIPTION
**Pull request by adhiravishankar**

### CODE REVIEW

**Comment by adhiravishankar** [comment]
There that should fix all the issues

**Comment by lfarah** [comment]
screen shot 2016-09-14 at 9 32 27 pm
There are errors....

**Comment by lfarah** [comment]
@adhiravishankar

File: Sources/DoubleExtensions.swift [hunk 1]
```diff
@@ -7,39 +7,33 @@
...
     }
-
+    
     /// EZSE: Returns a Double Ceil to decimal
-    public func getCeiledByPlaces(places: Int) -> Double {
-        return castToDecimalByPlacesHelper(places, function: ceil)
-    }
-
-    /// EZSE: Ceils current Double to number of places
-    public mutating func ceilByPlaces(places: Int) {
```
Comment (goktugyil) [inline Sources/DoubleExtensions.swift]: Why did you decide to delete this?

File: Sources/DoubleExtensions.swift [hunk 2]
```diff
@@ -7,39 +7,33 @@
...
-        self = castToDecimalByPlacesHelper(places, function: ceil)
-    }
-
-    private func castToDecimalByPlacesHelper(places: Int, function: (Double) -> Double) -> Double {
```
Comment (goktugyil) [inline Sources/DoubleExtensions.swift]: And why did you decide to delete this?

File: Sources/EZSwiftFunctions.swift [hunk 3]
```diff
@@ -23,17 +23,12 @@ public struct ez {
...
-        return NSBundle.mainBundle().objectForInfoDictionaryKey("CFBundleShortVersionString") as? String
+        return Bundle.main.object(forInfoDictionaryKey: "CFBundleShortVersionString") as? String
     }
 
     /// EZSE: Return app's build number
     public static var appBuild: String? {
-        return NSBundle.mainBundle().objectForInfoDictionaryKey(kCFBundleVersionKey as String) as? String
-    }
-
-    /// EZSE: Return app's bundle ID
```
Comment (goktugyil) [inline Sources/EZSwiftFunctions.swift]: And why did you decide to delete this?

File: Sources/EZSwiftFunctions.swift [hunk 4]
```diff
@@ -95,186 +90,141 @@ public struct ez {
...
+        let mainQueue = OperationQueue.main
+        NotificationCenter.default.addObserver(forName: NSNotification.Name.UIApplicationUserDidTakeScreenshot, object: nil, queue: mainQueue) { notification in
             // executes after screenshot
             action()
         }
     }
 
-    //TODO: Document this, add tests to this, find a way to remove ++
-    /// EZSE: Iterates through enum elements, use with (for element in ez.iterateEnum(myEnum))
-    public static func iterateEnum<T: Hashable>(_: T.Type) -> AnyGenerator<T> {
```
Comment (goktugyil) [inline Sources/EZSwiftFunctions.swift]: And why did you decide to delete this?

File: Sources/EZSwiftFunctions.swift [hunk 5]
```diff
@@ -95,186 +90,141 @@ public struct ez {
...
-        return AnyGenerator {
-            let next = withUnsafePointer(&i) { UnsafePointer<T>($0).memory }
-            return next.hashValue == i++ ? next : nil
-        }
-    }
-
     // MARK: - Dispatch
 
-    /// EZSE: Runs the function after x seconds
-    public static func dispatchDelay(second: Double, closure:()->()) {
```
Comment (goktugyil) [inline Sources/EZSwiftFunctions.swift]: And why did you decide to delete this?

File: Sources/IntExtensions.swift [hunk 6]
```diff
@@ -35,11 +35,8 @@ extension Int {
     /// EZSE: Converts integer value to UInt.
     public var toUInt: UInt { return UInt(self) }
 
-    /// EZSE: Converts integer value to Int32.
-    public var toInt32: Int32 { return Int32(self) }
```
Comment (goktugyil) [inline Sources/IntExtensions.swift]: Why delete this?

File: Sources/NSURLExtensions.swift [hunk 7]
```diff
@@ -25,249 +22,4 @@ extension NSURL {
 
         return parameters
     }
-
-    /// EZSE: Returns remote size of url, don't use it in main thread
-    public func remoteSize(completionHandler: ((contentLength: Int64) -> Void), timeoutInterval: NSTimeInterval = 30) {
```
Comment (goktugyil) [inline Sources/NSURLExtensions.swift]: Why delete this?

File: Sources/NSURLExtensions.swift [hunk 8]
```diff
@@ -25,249 +22,4 @@ extension NSURL {
...
-        NSURLSession.sharedSession().dataTaskWithRequest(request, completionHandler: { (_, response, _) -> Void in
-            let responseCode = (response as? NSHTTPURLResponse)?.statusCode ?? -1
-            dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), {
-                completionHandler(doesSupport: responseCode == 206)
-            })
-        }).resume()
-    }
-
-    /// EZSE: Compare two URLs
-    public func isSameWithURL(url: NSURL) -> Bool {
```
Comment (goktugyil) [inline Sources/NSURLExtensions.swift]: Why delete this?

File: Sources/StringExtensions.swift [hunk 9]
```diff
@@ -53,154 +53,23 @@ extension String {
...
     }
 
     /// EZSE: Capitalizes first character of String
-    public mutating func capitalizeFirst() {
-        guard characters.count > 0 else { return }
-        self.replaceRange(startIndex...startIndex, with: String(self[startIndex]).capitalizedString)
-    }
-
-    /// EZSE: Capitalizes first character of String, returns a new string
-    public func capitalizedFirst() -> String {
```
Comment (goktugyil) [inline Sources/StringExtensions.swift]: Why delete this?

**Pull request merged**